pkg/controller/kmeshsecurity/security_manager.go

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kmeshsecurity
+
+import (
+	"sync"
+	"time"
+
+	"container/heap"
+
+	"istio.io/istio/pkg/security"
+	"kmesh.net/kmesh/pkg/controller/workload"
+	"kmesh.net/kmesh/pkg/logger"
+)
+	
+type Cert struct {
+	uid string
+	exp time.Time
+}
+
+type PriorityQueue []*Cert
+
+var certs_maps sync.Map
+var log = logger.NewLoggerField("kmeshsecurity")
+
+type SecretManagerClient struct {
+	caClient *CitadelClient
+
+	// configOptions includes all configurable params for the cache.
+	configOptions *security.Options
+
+	// storing certificates
+	secretcache *sync.Map
+
+	caRootPath string
+
+	pending	PriorityQueue
+}
+ 
+var ScClient SecretManagerClient
+
+const (
+	rootCertPath       = "/var/run/secrets/istio/root-cert.pem"
+)
+   
+func (pq PriorityQueue) Len() int { return len(pq) }
+
+func (pq PriorityQueue) Less(i, j int) bool {
+	return pq[i].exp.Before(pq[j].exp)
+}
+
+func (pq PriorityQueue) Swap(i, j int) {
+	pq[i], pq[j] = pq[j], pq[i]
+}
+
+func (pq *PriorityQueue) Push(x interface{}) {
+	item := x.(*Cert)
+	*pq = append(*pq, item)
+}
+
+func (pq *PriorityQueue) Pop() interface{} {
+	old := *pq
+	n := len(old)
+	item := old[n-1]
+	*pq = old[0 : n-1]
+	return item
+}
+
+// NewSecretManagerClient creates a new SecretManagerClient.
+func NewSecretManagerClient() (*SecretManagerClient, error) {
+
+	tlsOpts = &TLSOptions{
+		RootCert:      rootCertPath,
+	}
+
+	options:= NewSecurityOptions()
+	caClient, err := NewCitadelClient(options, tlsOpts)
+	if err != nil {
+		return nil, err
+	}
+	pq := make(PriorityQueue, 0)
+	heap.Init(&pq)
+
+	ret := &SecretManagerClient{
+		caClient:      caClient,
+		configOptions: options,
+		caRootPath:  options.CARootPath,
+		secretcache: &certs_maps,
+		pending:	pq,
+	}
+	ScClient = *ret
+	go ScClient.delayedTask()
```

Comment:
Generally, it is recommended that the function behavior be consistent with the function name. You can return the created SecretManagerClient and run the function at the place where the function is called.